scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java

```diff
@@ -977,17 +978,37 @@ void pingTaskNode(RunningTaskData taskData) {
         } catch (Throwable t) {
             RunningTaskData runningTask = jobs.getRunningTask(task.getId());
             if (runningTask != null) {
-                int attempts = runningTask.increaseAndGetPingAttempts();
-                if (attempts > PASchedulerProperties.SCHEDULER_NODE_PING_ATTEMPTS.getValueAsInt()) {
-                    tlogger.info(task.getId(), "node failed", t);
-                    restartTaskOnNodeFailure(task);
+                // We try to ping the node where the task is running.
+                // We don't consider here other nodes reserved for the task, 
+                // as it is the responsibility of the task itself to manage extra nodes lifecycle 
+                // in case of complex multinodes task deployment.
+                if (!pingNode(runningTask.getNodes().get(0))) {
```

Comment:
It is the purpose of this pull request. The current behavior is to detect a node failure when there is an exception calling the getProgress method. However, this occurs regularly simply at the end of the task (the TaskLauncher autokills itself), without any real node failure. You can observe the difference by comparing the number of "node failed" message in this pull request tests:
http://jenkins.activeeon.com/view/Pull%20Requests/job/scheduling-pull-requests/1348/jdk=JDK8,label=Linux/artifact/logs/Scheduler-tests.log  => 1 "node failed" due to a dedicated test.

compared to the scheduling project logs:
http://jenkins.activeeon.com/job/scheduling/jdk=JDK8,label=Linux/lastSuccessfulBuild/artifact/logs/Scheduler-tests.log  => 4 "node failed"